source/extensions/access_loggers/open_telemetry/grpc_access_log_impl.cc

```diff
@@ -44,6 +35,34 @@ opentelemetry::proto::common::v1::KeyValue getStringKeyValue(const std::string&
 
 } // namespace
 
+GrpcAccessLoggerImpl::GrpcAccessLoggerImpl(
+    const Grpc::RawAsyncClientSharedPtr& client,
+    const envoy::extensions::access_loggers::open_telemetry::v3::OpenTelemetryAccessLogConfig&
+        config,
+    Event::Dispatcher& dispatcher, const LocalInfo::LocalInfo& local_info, Stats::Scope& scope)
+    : GrpcAccessLogger(
+          config.common_config(), dispatcher, scope, std::nullopt,
+          std::make_unique<Common::UnaryGrpcAccessLogClient<ExportLogsServiceRequest,
+                                                            ExportLogsServiceResponse>>(
+              client,
+              *Protobuf::DescriptorPool::generated_pool()->FindMethodByName(
+                  "opentelemetry.proto.collector.logs.v1.LogsService.Export"),
+              GrpcCommon::optionalRetryPolicy(config.common_config()), genOTelCallbacksFactory())),
+      stats_({ALL_GRPC_ACCESS_LOGGER_STATS(POOL_COUNTER_PREFIX(scope, GRPC_LOG_STATS_PREFIX))}) {
+  initMessageRoot(config, local_info);
+}
+
+std::function<GrpcAccessLoggerImpl::OTelLogRequestCallbacks*()>
+GrpcAccessLoggerImpl::genOTelCallbacksFactory() {
+  return [this]() {
+    OTelLogRequestCallbacks* ptr = new OTelLogRequestCallbacks(
+        this->stats_, this->batched_log_entries_,
+        [this](OTelLogRequestCallbacks* p) { this->callbacks_.erase(p); });
+    this->batched_log_entries_ = 0;
+    this->callbacks_.emplace(ptr, ptr);
```

Comment:
I think this should crash in non optimized builds. ptr is not a valid iterator position in callbacks_

Reply:
Hi Yan, I think `emplace(ptr, ptr)` treats the first argument as the key and the second argument to construct a unique_ptr as value. I don't know why you think it won't work?

Reply:
Oh sorry. I did not realize you were using hash map. Please disregard.